File: Backend/ECORUTAAS/src/main/java/com/cibertec/dtos/HistorialRutaDTO.java
package com.cibertec.dtos;

import java.util.List; // ✅ Import correcto
import com.cibertec.models.HistorialRuta.ModoTransporte;
import lombok.Data;

@Data
public class HistorialRutaDTO {

    private Integer idRuta;            // id de la ruta, si es ruta predefinida
    private String usuarioId;          // id del usuario que realiza la ruta
    private ModoTransporte modo;       // tipo de transporte seleccionado en el spinner
    private List<List<Double>> ruta;   // lista de coordenadas [lat, lng] de la ruta completa
    private Double distanciaRecorrida; // opcional: calculada en Android
    private Integer duracionMinutos;   // opcional: calculada en Android
}

File: Mobile/app/src/main/java/com/cibertec/proyectoecorutasapp/models/HistorialRutaDTO.kt
package com.cibertec.proyectoecorutasapp.models

data class HistorialRutaDTO(
    val idRuta: Int? = null,           // null para rutas libres
    val usuarioId: String,             // id del usuario (Firebase UID)
    val modo: String,                  // "BICICLETA", "SCOOTER", etc.
    val ruta: List<List<Double>>,      // [[lat,lng], [lat,lng], ...]
    val distanciaRecorrida: Double,
    val duracionMinutos: Int
)
